File: frontend/lib/models/Via.dart
class Via {
  final int id;
  final String via;
  final int? grau;
  final String? crux;
  final String? aid;
  final String? duracao;
  final String? exposicao;
  final int? extensao;
  final String? conquistadores;
  final String? data;
  final int mountId;
  final int faceId;
  final int? sourceId;
  final int? varianteId;

  Via({
    required this.id,
    required this.via,
    this.grau,
    this.crux,
    this.aid,
    this.duracao,
    this.exposicao,
    this.extensao,
    this.conquistadores,
    this.data,
    required this.mountId,
    required this.faceId,
    this.sourceId,
    this.varianteId,
  });
}

File: backend/src/entities/Via.ts
// entities/Via.ts

import {
    Entity,
    PrimaryGeneratedColumn,
    Column,
    ManyToOne,
    JoinColumn, CreateDateColumn, UpdateDateColumn
} from "typeorm";
import { Montanha } from "./Montanha";  // Assumindo que você também terá uma entidade Montanha
import { Face } from "./Face";          // Assumindo que você também terá uma entidade Face
import { Fonte } from "./Fonte";        // Assumindo que você também terá uma entidade Fonte
import { Variante } from "./Variante";


@Entity("vias")
export class Via {
    @PrimaryGeneratedColumn()
    id!: number;

    @Column({ type: "varchar", length: 255 })
    vias!: string;

    @Column({ type: "float", nullable: true })
    grau?: number;

    @Column({ type: "varchar", length: 50, nullable: true })
    crux?: string;

    @Column({ type: "varchar", length: 50, nullable: true })
    aid?: string;

    @Column({ type: "varchar", length: 50, nullable: true })
    duracao?: string;

    @Column({ type: "varchar", length: 50, nullable: true })
    exposicao?: string;

    @Column({ type: "int", nullable: true })
    extensao?: number;

    @Column({ type: "text", nullable: true })
    conquistadores?: string;

    @Column({ type: "date", nullable: true })
    data?: Date;

    @ManyToOne(() => Montanha)
    @JoinColumn({ name: "mount_id" })
    montanha!: Montanha;

    @ManyToOne(() => Face)
    @JoinColumn({ name: "face_id" })
    face!: Face;

    @ManyToOne(() => Fonte)
    @JoinColumn({ name: "source_id" })
    fonte!: Fonte;

    @ManyToOne(() => Variante)
    @JoinColumn({ name: "variante_id" })
    variante!: Variante;

    @CreateDateColumn()
    created_at!: Date;

    @UpdateDateColumn()
    updated_at!: Date;


}
